### PR TITLE
Adapt high-availability configuration

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area high-availability
/kind enhancement
/platform aws

**What this PR does / why we need it**:
This PR prepares components for HA by labeling them with the corresponding `high-availability-config.resources.gardener.cloud/type` and defining the required `PodDisruptionBudget`, please see [this document](https://github.com/gardener/gardener/blob/master/docs/development/high-availability.md) for more information.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/6529

**Special notes for your reviewer**:
- The Custom-Route-Controller cannot be adjusted yet due to missing leader election features, see https://github.com/gardener/aws-custom-route-controller/issues/2.
- [AWS LB Readadvertiser](https://github.com/gardener/aws-lb-readvertiser) misses leader election features as well, but can probably be neglected.

~Remains in draft mode until I gave it a test spin.~

**Release note**:

```feature operator
Extension and control-plane components have been adjusted to the high availability configuration supported by Gardener as of release `v1.60`. See [this document](https://github.com/gardener/gardener/blob/master/docs/development/high-availability.md) for more information about HA in gardener.
```
```feature user
This release adds HA support for AWS's control-plane components when extension is deployed along with Gardener >= `v1.60`. Please see [this document](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_high_availability.md) for more information about HA settings in shoot clusters.
```

### CODE REVIEW

File: charts/internal/seed-controlplane/charts/cloud-controller-manager/templates/deployment.yaml [hunk 1]
```diff
@@ -52,6 +53,7 @@ spec:
         - --authorization-kubeconfig=/var/run/secrets/gardener.cloud/shoot/generic-kubeconfig/kubeconfig
         - --leader-elect=true
         - --secure-port={{ include "cloud-controller-manager.port" . }}
+        - --leader-elect
```
Comment (kon-angelo) [inline charts/internal/seed-controlplane/charts/cloud-controller-manager/templates/deployment.yaml:56]: Isn't line `54` the equivalent ?

**Comment by timuthy** [comment]
Update: Changes have been tested and discussed with @kon-angelo. Ready for review.

File: charts/gardener-extension-provider-aws/templates/deployment.yaml [hunk 2]
```diff
@@ -9,6 +9,7 @@ metadata:
 {{- end }}
   labels:
 {{ include "labels" . | indent 4 }}
+    high-availability-config.resources.gardener.cloud/type: controller
```
Comment (timebertt) [inline charts/gardener-extension-provider-aws/templates/deployment.yaml:12]: Short question :)
This component is both a controller and a server.
I'm wondering, how to deal with such cases. Is the plan to always use `controller` in such cases?

Comment (rfranzke) [reply charts/gardener-extension-provider-aws/templates/deployment.yaml:12]: We should rather use `server` for such cases (even though, right now it doesn't make a difference technically, but in the future `server`s might have more replicas than controllers for better network traffic control).

Comment (timuthy) [reply charts/gardener-extension-provider-aws/templates/deployment.yaml]: Thanks for pointing this out. Makes sense 👍

Comment (timuthy) [reply charts/gardener-extension-provider-aws/templates/deployment.yaml]: FYI: I filed https://github.com/gardener/gardener/pull/7060 to include this lessons learned.

**Comment by kon-angelo** [comment]
@timuthy Are there any additional changes that are needed ?

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**